Skin menu selection

Codidodido opened the pull request:
- Create skin menu
- Select skin

Codidodido committed:
Add monogram font

Codidodido committed:
Create skin card class and skins scriptable object

Codidodido committed:
Create change frame script

Create change frame script which changes frame when clicked on

Codidodido committed:
Create skin card displayer script

Create skin card displayer script which create cards from skin scriptable object

Codidodido committed:
Create skin card controller script

Create skin card controller script which handles card view and skin selection

Codidodido committed:
Create Card Prefab

Codidodido committed:
Create skin asset from skins scriptable object

Create skin asset from skins scriptable object to store skins

Codidodido committed:
Create skin shop scene

URAmiRBin commented on Assets/Scripts/ChangeFrame.cs:
```diff
@@ -0,0 +1,23 @@
+    public void IncreaseFrame()
+    {
+        onIncreaseFrame?.Invoke();
+        Debug.Log("Increase");
```
remove the log messages

URAmiRBin commented on Assets/Scripts/ChangeFrame.cs:
```diff
@@ -0,0 +1,23 @@
+    public void DecreaseFrame()
+    {
+        onDecreaseFrame?.Invoke();
+        Debug.Log("Decrease");
```
also this

URAmiRBin commented on Assets/Scripts/ChangeFrame.cs:
```diff
@@ -0,0 +1,23 @@
+
+    public void IncreaseFrame()
+    {
+        onIncreaseFrame?.Invoke();
```
find a way to remove the null check

Codidodido replied:
Why ?
Maybe I find solution easier and faster If I know the reason

URAmiRBin replied:
null checks are bad code smell if you have learned your lessons well

URAmiRBin commented on Assets/Scripts/SkinCardController.cs:
```diff
@@ -0,0 +1,24 @@
+    public void SelectSkin()
+    {
+        PlayerPrefs.SetString("skin",cardSkin.sprite.name);
+        Debug.Log(cardSkin.sprite.name);
```
no logs, just use debugger for test

URAmiRBin commented on Assets/Scripts/SkinCardDisplayer.cs:
```diff
@@ -0,0 +1,87 @@
+    [SerializeField] private ChangeFrame increaseFrame;
+    [SerializeField] private ChangeFrame decreaseFrame;
+    
+    public int frame;
```
frame is not an integer

URAmiRBin commented on Assets/Scripts/ChangeFrame.cs:
```diff
@@ -0,0 +1,23 @@
+using System.Security.Cryptography;
+using UnityEngine;
+
+public class ChangeFrame : MonoBehaviour
```
ChangeFrame is not a good name for this class

URAmiRBin commented on Assets/Scripts/SkinCardDisplayer.cs:
```diff
@@ -0,0 +1,87 @@
+        RemoveCards();
+        for (var i = frame; i < frame + skinCountPerRow; i++)
+        {
+            if(i<skins.SkinCards.Count)
```
```suggestion
            if (i < skins.SkinCards.Count)
```

for readablity

URAmiRBin commented on Assets/Scripts/SkinCardDisplayer.cs:
```diff
@@ -0,0 +1,87 @@
+        
+        card.GetComponent<SkinCardController>().SetCard(skinCard.skinSprite,skinCard.name);
+        
+        if(skinCard.skinSprite.name == PlayerPrefs.GetString("skin"))
```
make this boolean a function, what does this do?

Codidodido replied:
Nothing That much important, I wanted to display selected card with red color, but I have to find better solution

URAmiRBin replied:
ok find a better solution

URAmiRBin commented on Assets/Scripts/SkinCardDisplayer.cs:
```diff
@@ -0,0 +1,87 @@
+        var cards = cardsGroup.GetComponentsInChildren<SkinCardController>();
+        
+        foreach (var card in cards)
+        {
+            Destroy(card.gameObject);
+        }
```
```suggestion
            Destroy(card.gameObject);
```

URAmiRBin commented on Assets/Scripts/ChangeFrame.cs:
```diff
@@ -0,0 +1,23 @@
+    public Action onIncreaseFrame;
+    public Action onDecreaseFrame;
+
+    public void IncreaseFrame()
```
what does this mean??

URAmiRBin commented:
@Codidodido 
dont mark the conversations as resolved if you have not done any changes

Codidodido committed:
Refactor skin handler scripts

- Rename change frame script to start skin card index changer script and remove debug log and null check of actions
- remove debug log from skin card controller and skin card displayer
- Implement name changes, Add spaces and remove useless lines from skin card displayer

Codidodido committed:
Add current skin card displayer and edit other scripts

Add current skin card displayer and edit other scripts to display selected skin in scene

URAmiRBin commented on Assets/Scripts/StartSkinCardIndexChanger.cs:
```diff
@@ -0,0 +1,21 @@
+using System.Security.Cryptography;
+using UnityEngine;
+
+public class StartSkinCardIndexChanger : MonoBehaviour
```
What does this do the name does not ring any bells to me

URAmiRBin commented on Assets/Scripts/StartSkinCardIndexChanger.cs:
```diff
@@ -0,0 +1,21 @@
+
+public class StartSkinCardIndexChanger : MonoBehaviour
+{
+    public Action onIncreaseFrame = () => { };
```
what does this mean? how do you increase frames?

URAmiRBin commented:
@Codidodido reviewed, still need some changes

Codidodido committed:
Refactor

- Rename StartSkinCardIndexChanger script to ScrollPreviewNavigator
- Rename CurrentSkinCardDisplayer script to SkinPreviewNavigator
- Rename onIncreaseFrame & onDecreaseFrame Action to onForwardScroll & onBackwardScroll in ScrollPreviewNavigator
- Rename IncreaseFrame & DecreaseFrame methods to ForwardScroll & BackwardScroll in SkinCardDisplayer
